analysis/src/TypeUtils.ml

```diff
@@ -178,6 +328,171 @@ let rec extractType ~env ~package (t : Types.type_expr) =
     | _args, _tRet -> None)
   | _ -> None
 
+let rec extractType2 ?(printOpeningDebug = true)
```

Comment:
This can't fully replace `extractType` yet because that fn is used internally here some. Will need to be fully replaced in separate refactor.